### PR TITLE
[Form] fix support for years outside of the 32b range on x86 arch

### PR DESCRIPTION
**Pull request by nicolas-grekas**
| Q             | A
| ------------- | ---
| Branch?       | 4.4
| Bug fix?      | yes
| New feature?  | no
| Deprecations? | no
| Tickets       | -
| License       | MIT
| Doc PR        | -

### CODE REVIEW

File: src/Symfony/Component/Security/Http/Tests/RememberMe/PersistentRememberMeHandlerTest.php [hunk 1]
```diff
@@ -118,7 +118,7 @@ public function testConsumeRememberMeCookieExpired()
         $this->tokenProvider->expects($this->any())
             ->method('loadTokenBySeries')
             ->with('series1')
-            ->willReturn(new PersistentToken(InMemoryUser::class, 'wouter', 'series1', 'tokenvalue', new \DateTime('-'.(31536000 - 1).' years')));
+            ->willReturn(new PersistentToken(InMemoryUser::class, 'wouter', 'series1', 'tokenvalue', new \DateTime('-'.(31536000 - 1).' months')));
```
Comment (nicolas-grekas) [inline src/Symfony/Component/Security/Http/Tests/RememberMe/PersistentRememberMeHandlerTest.php]: /cc @wouterj WDYT?

File: src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php [hunk 2]
```diff
@@ -194,11 +194,15 @@ public function format($timestamp)
     {
         // intl allows timestamps to be passed as arrays - we don't
         if (\is_array($timestamp)) {
-            $message = 'Only integer Unix timestamps and DateTime objects are supported';
+            $message = 'Only Unix timestamps and DateTime objects are supported';
 
             throw new MethodArgumentValueNotImplementedException(__METHOD__, 'timestamp', $timestamp, $message);
         }
 
+        if (\is_string($timestamp) && $dt = \DateTime::createFromFormat('U', $timestamp)) {
```
Comment (stof) [inline src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php:200]: This should also be applied to the polyfills repo, right ?

Comment (nicolas-grekas) [reply src/Symfony/Component/Intl/DateFormatter/IntlDateFormatter.php:200]: yep, see https://github.com/symfony/polyfill/pull/357

File: src/Symfony/Component/Security/Http/Tests/RememberMe/PersistentRememberMeHandlerTest.php (hunk 1 above)
Comment (wouterj) [reply src/Symfony/Component/Security/Http/Tests/RememberMe/PersistentRememberMeHandlerTest.php]: To me, seems like we can even make this `'-'.(31536000 + 1).' seconds'`?

**Review by stof — APPROVED** [review]

**Pull request merged**